client/src/test/java/com/netflix/conductor/client/automator/TaskPollExecutorTest.java

```diff
@@ -166,14 +164,45 @@ public void testTaskPollException() {
         Worker worker = mock(Worker.class);
         when(worker.getPollingInterval()).thenReturn(3000);
         when(worker.getTaskDefName()).thenReturn("test");
-        when(worker.preAck(any())).thenReturn(true);
         when(worker.execute(any())).thenReturn(new TaskResult(task));
 
         TaskClient taskClient = Mockito.mock(TaskClient.class);
         when(taskClient.pollTask(any(), any(), any()))
             .thenThrow(ConductorClientException.class)
             .thenReturn(task);
-        when(taskClient.ack(any(), any())).thenReturn(true);
+
+        TaskPollExecutor taskPollExecutor = new TaskPollExecutor(null, taskClient, 1, 1, "test-worker-");
+        CountDownLatch latch = new CountDownLatch(1);
+        doAnswer(invocation -> {
+                Object[] args = invocation.getArguments();
+                TaskResult result = (TaskResult) args[0];
+                assertEquals(IN_PROGRESS, result.getStatus());
+                assertEquals(task.getTaskId(), result.getTaskId());
+                latch.countDown();
+                return null;
+            }
+        ).when(taskClient).updateTask(any());
+
+        Executors.newSingleThreadScheduledExecutor()
+            .scheduleAtFixedRate(() -> taskPollExecutor.pollAndExecute(worker), 0, 1, TimeUnit.SECONDS);
+
+        Uninterruptibles.awaitUninterruptibly(latch);
+        verify(taskClient).updateTask(any());
+    }
+
+    @Test
+    public void testTaskPoll() {
+        Task task = testTask();
+
+        Worker worker = mock(Worker.class);
+        when(worker.getPollingInterval()).thenReturn(3000);
+        when(worker.getTaskDefName()).thenReturn("test");
+        when(worker.execute(any())).thenReturn(new TaskResult(task));
+
+        TaskClient taskClient = Mockito.mock(TaskClient.class);
+        when(taskClient.pollTask(any(), any(), any()))
+            .thenReturn(new Task())
+            .thenReturn(task);
```

Comment:
Is it intentional to have two `thenReturn` statements?

Reply:
yes, these will be the values returned by this method on 2 consecutive invocations.